src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor_state.go

```diff
@@ -9,17 +11,28 @@ import (
 )
 
 func (st *StoragePowerActorState_I) ActivePowerMeetsConsensusMinimum(minerPower block.StoragePower) bool {
-	totPower := st._getActivePower()
-
 	// TODO import from consts
 	MIN_MINER_SIZE_STOR := block.StoragePower(0)
-	MIN_MINER_SIZE_PERC := 0
+	MIN_MINER_SIZE_TARG := 0
+
+	// if a miner meets min power requirement check whether this one does
+	if st._minersLargerThanMin() > util.UVarint(0) {
+		return minerPower >= MIN_MINER_SIZE_STOR
+	}
+
+	// else check whether in MIN_MINER_SIZE_TARG miners
+	if len(st.PowerTable()) < MIN_MINER_SIZE_TARG {
+		// miner should pass
+		return true
+	}
 
-	// if miner smaller than both min size in bytes and min percentage
-	if (int(minerPower)*MIN_MINER_SIZE_PERC < int(totPower)*100) && minerPower < MIN_MINER_SIZE_STOR {
-		return false
+	// get size of MIN_MINER_SIZE_TARGth largest miner
+	minerSizes := make([]block.StoragePower, 0, len(st.PowerTable()))
+	for _, v := range st.PowerTable() {
+		minerSizes = append(minerSizes, v.ActivePower())
 	}
-	return true
+	sort.Slice(minerSizes, func(i, j int) bool { return int(i) < int(j) })
```

Comment:
```suggestion
	sort.Slice(minerSizes, func(i, j int) bool { return int(i) > int(j) })
```
... for descending order